### PR TITLE
Rrivera/post gr2checkpoint

### PR DESCRIPTION
**Pull request by rrivera747**

### CODE REVIEW

**Comment by rrivera747** [comment]
@antoniogioiosa I reorganized the runPlugin handling. Hopefully, you agree it is still the same behavior. I have one main question, why does the runPlugin need the full config dump blob on the Run Transition?

**Comment by antoniogioiosa** [comment]
Hi Ryan,
Runinfo needs the blob to fill the run_configuration table with information
like the configuration name and version and the context name and version.
It would be better for me if they could be available at this level (without
the blob).
Thanks
Antonio

Il giorno mar 15 ott 2024 alle ore 15:24 rrivera747 <
***@***.***> ha scritto:

**Comment by rrivera747** [comment]
Ok, they are definitely available. Maybe you can simplify the run transition parameters in the future?

Thanks,
Ryan


________________________________

**Comment by antoniogioiosa** [comment]
I hope,
we can discuss again a better version of the run_configuration table.

Thanks
Antoino

Il giorno mar 15 ott 2024 alle ore 16:01 rrivera747 <
***@***.***> ha scritto:

**Comment by rrivera747** [comment]
Yes, let's discuss the details soon. Thanks!

Ryan


________________________________

File: otsdaq/CodeEditor/CodeEditor.cc [hunk 1]
```diff
@@ -83,6 +84,11 @@ try
 	{
 		xmlOut->addTextElementToData("AllowedExtensions", StringMacros::setToString(ALLOWED_FILE_EXTENSIONS_, ","));
 	}
+	else if(option == "getFileGitURL")
+	{
+		//TODO!
```
Comment (eflumerf) [inline otsdaq/CodeEditor/CodeEditor.cc]: What's the TODO? Looks implemented to me...

File: otsdaq/CodeEditor/CodeEditor.cc [hunk 2]
```diff
@@ -380,6 +386,60 @@ void CodeEditor::getFileContent(cgicc::Cgicc& cgiIn, HttpXmlDocument* xmlOut)
...
+
+}  // end getFileGitURL()
+
+//==============================================================================
+// getFileGitURL
+std::string CodeEditor::getFileGitURL(const std::string& basepath, const std::string& path)
+{
+
+		//TODO!
+		//TODO! request with linux exec to spack environment
```
Comment (eflumerf) [inline otsdaq/CodeEditor/CodeEditor.cc:426]: Can we have Name/Date with TODO comments so we can tell how stale they are? Even better would be a Github issue number/link

File: otsdaq/ConfigurationInterface/ConfigurationManager.cc [hunk 3]
```diff
@@ -3487,6 +3521,432 @@ bool ConfigurationManager::isOwnerFirstAppInContext()
...
+		if(invertFilter)
+			++i;
+		size_t      f;
+		std::string tmp;
+		while((f = stateMachineAliasFilter.find('*', i)) != std::string::npos)
+		{
+			tmp = stateMachineAliasFilter.substr(i, f - i);
+			i   = f + 1;
+			filterArr.push_back(tmp);
+			//__COUT__ << filterArr[filterArr.size()-1] << " " << i <<
```
Comment (eflumerf) [inline otsdaq/ConfigurationInterface/ConfigurationManager.cc]: Make a TRACE or delete

File: otsdaq/ConfigurationInterface/ConfigurationManager.cc [hunk 4]
```diff
@@ -3487,6 +3521,432 @@ bool ConfigurationManager::isOwnerFirstAppInContext()
...
+			//__COUT__ << "filterMatch=" << filterMatch  << __E__;
+
+			if(!filterMatch)
+				continue;
+
+			retSet.emplace(aliasMapPair.first);
+
+			//bring back below, if need more alias details:
+
+			// xmlOut.addTextElementToData("config_alias", aliasMapPair.first);
```
Comment (eflumerf) [inline otsdaq/ConfigurationInterface/ConfigurationManager.cc]: Maybe a #if EXTENDED_ALIAS_DETAILS ?

File: otsdaq/ConfigurationInterface/Database_configInterface.cc [hunk 5]
```diff
@@ -34,7 +34,7 @@ constexpr auto default_entity     = "OTSROOT";
 //==============================================================================
 DatabaseConfigurationInterface::DatabaseConfigurationInterface()
 {
-#ifdef DEBUG_ENABLE
+// #ifdef DEBUG_ENABLE
```
Comment (eflumerf) [inline otsdaq/ConfigurationInterface/Database_configInterface.cc]: I think I'd prefer the #ifdef over the if(0), if it's conflicting make it ARTDAQ_DATABASE_DEBUG_ENABLE

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc [hunk 6]
```diff
@@ -1161,226 +2145,333 @@ try
...
 				}
-				catch(...){}
-				__SS_THROW__;
-			}  // End write run info into db
+
+				//FIXME -- Should this 2nd param be activeStateMachineConfigurationDumpOnConfigure_?! What is the 2nd param for? Is conditionID_ enough?
+				runNumber = runInfoInterface->claimNextRunNumber(conditionID_, activeStateMachineConfigurationDumpOnRun_);
+			}  // end Run Info Plugin handling
+
+			// std::stringstream dumpSs;
```
Comment (eflumerf) [inline otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: Delete or ifdef out

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc [hunk 7]
```diff
@@ -1873,222 +2964,143 @@ try
...
-	{
-		__SS__ << "\nTransition to Configuring interrupted! System Alias " << systemAlias << " could not be translated to a group name and key." << __E__;
-
-		__COUT_ERR__ << "\n" << ss.str();
-		XCEPT_RAISE(toolbox::fsm::exception::Exception, ss.str());
-		return;
-	}
-
-	RunControlStateMachine::theProgressBar_.step();
+	// try
```
Comment (eflumerf) [inline otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: Delete or ifdef out

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc [hunk 8]
```diff
@@ -1873,222 +2964,143 @@ try
...
-			__COUT_INFO__ << "No Gateway Supervisor configuration record found at '" << ConfigurationManager::XDAQ_CONTEXT_TABLE_NAME << "/"
-			              << supervisorContextUID_ << "/" << supervisorApplicationUID_
-			              << "' - consider adding one to control configuration dumps and state machine properties." << __E__;
+			ss << " No user log entry.";
+		makeSystemLogEntry(ss.str());
+	}  // end make logbook entry
+
+	RunControlStateMachine::theProgressBar_.step();
+
+	// // load and activate
```
Comment (eflumerf) [inline otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: Delete or ifdef out

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc [hunk 9]
```diff
@@ -2124,20 +3136,350 @@ try
...
+			ss << "Exception message: " << e.what();
+		}
+		catch(...){}
+		__SS_THROW__;
+	}  // End write run condition into db
 	RunControlStateMachine::theProgressBar_.step();
-	// Advertise the exiting of this method
-	// diagService_->reportError("GatewaySupervisor::stateConfiguring: Exiting",DIAGINFO);
 
+	// // Check if Run Plugin is defined and, if so, create a new condition record into database
```
Comment (eflumerf) [inline otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: Delete or ifdef out

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc [hunk 10]
```diff
@@ -2548,110 +3918,187 @@ try
...
-				XCEPT_RAISE(toolbox::fsm::exception::Exception, ss.str());
-				return;
-			}
-		}
-		else
-			__COUT_INFO__ << "No Gateway Supervisor configuration record found at '" << ConfigurationManager::XDAQ_CONTEXT_TABLE_NAME << "/"
-			              << supervisorContextUID_ << "/" << supervisorApplicationUID_
-			              << "' - consider adding one to control configuration dumps and state machine properties." << __E__;
-	}
+	// check if configuration dump is enabled on configure transition
```
Comment (eflumerf) [inline otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: Delete or ifdef out

File: otsdaq/ConfigurationInterface/ConfigurationManager.cc (hunk 4 above)
Comment (rrivera747) [reply otsdaq/ConfigurationInterface/ConfigurationManager.cc]: deleted

File: otsdaq/ConfigurationInterface/Database_configInterface.cc (hunk 5 above)
Comment (rrivera747) [reply otsdaq/ConfigurationInterface/Database_configInterface.cc]: back to ifdef

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc (hunk 6 above)
Comment (rrivera747) [reply otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: deleted

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc (hunk 7 above)
Comment (rrivera747) [reply otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: deleted

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc (hunk 9 above)
Comment (rrivera747) [reply otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: deleted

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc (hunk 8 above)
Comment (rrivera747) [reply otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: deleted

File: otsdaq/GatewaySupervisor/GatewaySupervisor.cc (hunk 10 above)
Comment (rrivera747) [reply otsdaq/GatewaySupervisor/GatewaySupervisor.cc]: deleted

**Review by eflumerf — APPROVED** [review]

**Pull request merged**